July 28 releases blog (#442)
pulumi/pulumi-hugo: themes/default/content/blog/pulumi-release-notes-m59/index.md

---
title: "July 28 releases: K8s GitOps, autonaming in Google Native"
date: 2021-07-28T08:00:00-07:00
draft: false
meta_desc: "Kubernetes is faster, K8s operator supports GitOps, new resources and functionality for Azure and Google Cloud, and easier getting started in the Pulumi Service"
meta_image: meta.png
authors:
    - alex-mullans
tags:
    - features
---

> Another 3 weeks have passed, so it's time for another edition of the Pulumi release notes! This iteration we've been busy smashing bugs across our products (over 100!), but we've still got some great new features to share with you as well:

Reviewer: ```suggestion
Another great iteration has just wrapped up, and we've got several new updates across Pulumi providers, CLI and service to share:
```

Author: Overall I like your re-word, but any reason you took out the bug smashing item? I think it's important to note to the community that we're fixing small things, not just always building new stuff.

Author: ```suggestion
Another great iteration has just wrapped up, so it's time for another edition of the Pulumi release notes! We've been busy smashing bugs across our products (over 100!), but we've also got several new updates across Pulumi providers, the CLI, and the Pulumi Service:
```

- New and updated cloud providers

>   - [Kubernetes operator supports GitOps workflow]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-operator-supports-gitops-workflow" >}})

Reviewer: ```suggestion
  - [Pulumi Kubernetes Operator support for GitOps workflow]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-operator-supports-gitops-workflow" >}})
```

Author: ```suggestion
  - [Pulumi Kubernetes Operator supports GitOps workflow]({{< relref "/blog/pulumi-release-notes-m59#pulumi-kubernetes-operator-supports-gitops-workflow" >}})
```

>   - [Kubernetes improved performance]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-improved-performance" >}})

Reviewer: ```suggestion
  - [Kubernetes provider performance improvements]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-improved-performance" >}})
```

Author: ```suggestion
```

  - [12 new resources in the Azure Native provider]({{< relref "/blog/pulumi-release-notes-m59#12-new-resources-in-the-azure-native-provider" >}})
  - [Google Native: autonaming; easier management of project and region settings]({{< relref "/blog/pulumi-release-notes-m59#google-native-autonaming-easier-management-of-project-and-region-settings" >}})
- Pulumi CLI and core technologies
  - [Automation API: Select your backend]({{< relref "/blog/pulumi-release-notes-m59#automation-api-select-your-backend" >}})
- Pulumi Service and Pulumi.com
  - [Get started with Pulumi, directly from the Pulumi Service]({{< relref "/blog/pulumi-release-notes-m59#get-started-with-pulumi-directly-from-the-pulumi-service" >}})

<!--more-->

## New and updated cloud providers

> ### Kubernetes operator supports GitOps workflow

Reviewer: ```suggestion
### Pulumi Kubernetes Operator supports GitOps workflow
```

> For some Kubernetes workflows, developers want to be able to store their configuration in a Git repository but ensure that their cluster automatically stays up-to-date with the latest configuration when it's checked in. With the latest update to the Pulumi Kubernetes Operator, you can set up this type of GitOps workflow. To get started, [specify the fully-qualified branch name](https://github.com/pulumi/pulumi-kubernetes-operator/blob/80398a85958215a7c2c87e9ce30c69998f6cdba9/pkg/apis/pulumi/v1alpha1/stack_types.go#L99-L101) that you want to track in the `Stack` CustomResource (e.g., `refs/heads/main` will track the `main` branch). The operator will track the specified branch and automatically deploy new commits that are pushed to the branch.

Reviewer: ```suggestion
For some Kubernetes workflows, developers want to be able to store their configuration in a Git repository but ensure that their cluster automatically stays up-to-date with the latest configuration when it's checked in. With the latest update to the [Pulumi Kubernetes Operator](https://github.com/pulumi/pulumi-kubernetes-operator), you can set up this type of GitOps workflow. To get started, [specify the fully-qualified branch name](https://github.com/pulumi/pulumi-kubernetes-operator/blob/80398a85958215a7c2c87e9ce30c69998f6cdba9/pkg/apis/pulumi/v1alpha1/stack_types.go#L99-L101) that you want to track in the `Stack` CustomResource (e.g., `refs/heads/main` will track the `main` branch). The operator will track the specified branch and automatically deploy new commits that are pushed to the branch.
```

> [Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes-operator/issues/50)

Reviewer: Perhaps add a code snippet of a Stack that uses this?

The Kubernetes YAML below will tell the Pulumi Kubernetes Operator to poll the `production` branch of the target GitHub repo, and deploy an update to the `acmecorp/s3-op-project/production` stack when there are changes.

```yaml
apiVersion: pulumi.com/v1alpha1
kind: Stack
metadata:
  name: s3-bucket-stack
spec:
  stack: acmecorp/s3-op-project/production
  projectRepo: https://github.com/acmecorp/test-s3-op-project
  branch: production 
  config:
    aws:region: us-east-2
```

> [Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes-operator/issues/50)

Author: ```suggestion
For some Kubernetes workflows, developers want to be able to store their configuration in a Git repository but ensure that their cluster automatically stays up-to-date with the latest configuration when it's checked in. With the latest update to the Pulumi Kubernetes Operator, you can set up this type of GitOps workflow. To get started, [specify the fully-qualified branch name](https://github.com/pulumi/pulumi-kubernetes-operator/blob/80398a85958215a7c2c87e9ce30c69998f6cdba9/pkg/apis/pulumi/v1alpha1/stack_types.go#L99-L101) that you want to track in the `Stack` CustomResource (e.g., `refs/heads/main` will track the `main` branch). The operator will track the specified branch and automatically deploy new commits that are pushed to the branch. For example, the YAML snippet below will tell the Pulumi Kubernetes Operator to poll the production branch of the target GitHub repo, and deploy an update to the acmecorp/s3-op-project/production stack when there are changes:

```yaml
apiVersion: pulumi.com/v1alpha1
kind: Stack
metadata:
  name: s3-bucket-stack
spec:
  stack: acmecorp/s3-op-project/production
  projectRepo: https://github.com/acmecorp/test-s3-op-project
  branch: production 
  config:
    aws:region: us-east-2
```

[Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes-operator/issues/50)
```

### 12 new resources in the Azure Native provider

We shipped 4 new versions of the Azure Native provider (1.16.0 through 1.19.0) that collectively added 12 new resources that you can manage with the Azure Native provider, including Fast Healthcare Interoperability Resources (FHIR) and Azure Relay resources.

[See the full list](https://github.com/pulumi/pulumi-azure-native/blob/master/CHANGELOG.md#1190-2021-07-22)

### Google Native: autonaming; easier management of project and region settings

The preview of the Google Native provider continues to evolve. We shipped a new version of the Google Native provider ([0.5.0](https://github.com/pulumi/pulumi-google-native/releases/tag/v0.5.0)) that includes:

- Full auto-naming support, just like our other providers
- An easier way to use arguments like `machineType` that expect a fully-qualified string like `projects/${project}/zones/${zone}/machineTypes/f1-micro`: you can now use just the last value (`f1-micro`) in your program and the provider will construct the fully-qualified string
- Automatic selection of the project, region, and zone based on your configuration, so you no longer need to pass these values to the SDK's constructor

> - An improved implementation of `BucketObject`

Reviewer: Perhaps add a code snippet that shows off a few/all of these?

Author: Yes, good idea. @mikhailshilkov do you happen to have any snippets handy, or suggestions of ones I should author?

[See the full list of updates](https://github.com/pulumi/pulumi-google-native/blob/master/CHANGELOG.md)

## Pulumi CLI and core technologies

> In this milestone, we shipped Pulumi versions [3.7.0](https://github.com/pulumi/pulumi/releases/tag/v3.7.0), [3.7.1](https://github.com/pulumi/pulumi/releases/tag/v3.7.1), and [3.8.0](https://github.com/pulumi/pulumi/releases/tag/v3.8.0). The full list of changes in each version is available in the linked changelog; read on to learn about some of the biggest changes.

Reviewer: Perhaps worth expanding on here or in their own section:
* https://github.com/pulumi/pulumi/pull/7483
* Resource Methods

Author: Methods will get its own blog post in M60, as well as a shoutout here. I'll add the `Output[T]` one though.

- New and updated cloud providers
  - [Kubernetes operator supports GitOps workflow]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-operator-supports-gitops-workflow" >}})
  - [Kubernetes improved performance]({{< relref "/blog/pulumi-release-notes-m59#kubernetes-improved-performance" >}})
  - [12 new resources in the Azure Native provider]({{< relref "/blog/pulumi-release-notes-m59#12-new-resources-in-the-azure-native-provider" >}})
  - [Google Native: autonaming; easier management of project and region settings]({{< relref "/blog/pulumi-release-notes-m59#google-native-autonaming-easier-management-of-project-and-region-settings" >}})
- Pulumi CLI and core technologies
  - [Automation API: Select your backend]({{< relref "/blog/pulumi-release-notes-m59#automation-api-select-your-backend" >}})
- Pulumi Service and Pulumi.com
  - [Get started with Pulumi, directly from the Pulumi Service]({{< relref "/blog/pulumi-release-notes-m59#get-started-with-pulumi-directly-from-the-pulumi-service" >}})

<!--more-->

## New and updated cloud providers

### Kubernetes operator supports GitOps workflow

> For some Kubernetes workflows, developers want to be able to store their configuration in a Git repository but ensure that their cluster automatically stays up-to-date with the latest configuration when it's checked in. With the latest update to the Pulumi Kubernetes Operator, you can set up this type of GitOps workflow. To get started, TODO

Reviewer 2: ```suggestion
For some Kubernetes workflows, developers want to be able to store their configuration in a Git repository but ensure that their cluster automatically stays up-to-date with the latest configuration when it's checked in. With the latest update to the Pulumi Kubernetes Operator, you can set up this type of GitOps workflow. To get started, [specify the fully-qualified branch name](https://github.com/pulumi/pulumi-kubernetes-operator/blob/80398a85958215a7c2c87e9ce30c69998f6cdba9/pkg/apis/pulumi/v1alpha1/stack_types.go#L99-L101) that you want to track in the `Stack` CustomResource (e.g., `refs/heads/main` will track the `main` branch). The operator will track the specified branch and automatically deploy new commits that are pushed to the branch.
```

[Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes-operator/issues/50)

### Kubernetes improved performance

> TODO

Author: @lblackstone on Monday, can you help me draft content for this item (and the other TODO a few lines up)?

Reviewer 2: @infin8x The informer changes in this release aren't performance fixes; we're planning to work on that further later. The current changes make the provider more resilient to network flakiness, especially under heavy load. It's more of a correctness fix that most users probably won't notice. If you like, I can draft something up about it, or we can drop this section from the blog.

Author: Will drop this section, thanks for letting me know!

> ### Kubernetes improved performance
> 
> TODO
> 
> [Learn more in this GitHub issue](https://github.com/pulumi/pulumi-kubernetes/issues/1598)
> 

Author: ```suggestion
```

```yaml
apiVersion: pulumi.com/v1alpha1
kind: Stack
metadata:
  name: s3-bucket-stack
spec:
  stack: acmecorp/s3-op-project/production
  projectRepo: https://github.com/acmecorp/test-s3-op-project

>   branch: production 

Reviewer 2: ```suggestion
  branch: refs/heads/production 
```

Reviewer 2: See https://github.com/pulumi/pulumi-kubernetes-operator/issues/103#issuecomment-886932753

date: 2021-07-28T08:00:00-07:00
draft: false
meta_desc: "Kubernetes is faster, K8s operator supports GitOps, new resources and functionality for Azure and Google Cloud, and easier getting started in the Pulumi Service"
meta_image: meta.png
authors:
    - alex-mullans
tags:
    - features
---

Another great iteration has just wrapped up, so it's time for another edition of the Pulumi release notes! We've been busy smashing bugs across our products (over 100!), but we've also got several new updates across Pulumi providers, the CLI, and the Pulumi Service:

- New and updated cloud providers
  - [Pulumi Kubernetes Operator supports GitOps workflow]({{< relref "/blog/pulumi-release-notes-m59#pulumi-kubernetes-operator-supports-gitops-workflow" >}})
  - [12 new resources in the Azure Native provider]({{< relref "/blog/pulumi-release-notes-m59#12-new-resources-in-the-azure-native-provider" >}})
  - [Google Native: autonaming; easier management of project and region settings]({{< relref "/blog/pulumi-release-notes-m59#google-native-autonaming-easier-management-of-project-and-region-settings" >}})
- Pulumi CLI and core technologies

>   - [Automation API: Select your backend]({{< relref "/blog/pulumi-release-notes-m59#automation-api-select-your-backend" >}})

Author: ```suggestion
  - [Automation API: Select your backend]({{< relref "/blog/pulumi-release-notes-m59#automation-api-select-your-backend" >}})
  - [Better handling of subtypes in Python]({{< relref "/blog/pulumi-release-notes-m59#better-handling-of-subtypes-in-python" >}})
```

```yaml
apiVersion: pulumi.com/v1alpha1
kind: Stack
metadata:
  name: s3-bucket-stack
spec:
  stack: acmecorp/s3-op-project/production
  projectRepo: https://github.com/acmecorp/test-s3-op-project

>   branch: refs/head/production

Reviewer 2: ```suggestion
  branch: refs/heads/production
```
